presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java

```diff
@@ -478,28 +485,75 @@ private void testRoundTripType(Type type, List<?> readValues)
+            // Keeping types the same so that we can
+            // 1. reuse the "filters" variable;
+            List<Map<Integer, Map<Subfield, TupleDomainFilter>>> newFilters = filters.stream()
+                    .map(filter -> Maps.transformEntries(filter,
+                            (column, value) -> Maps.transformValues(value,
+                                    tupleDomainFilter -> addOrderTracking(tupleDomainFilter, orderChecker, column))))
+                    .collect(toImmutableList());
+            optionalOrderChecker = Optional.of(orderChecker);
+            filters = newFilters;
+        }
 
         // forward order
-        assertRoundTrip(types, readValues, filters);
+        assertRoundTrip(types, readValues, filters, optionalOrderChecker);
 
         // reverse order
         if (reverseTestsEnabled) {
-            assertRoundTrip(types, Lists.transform(readValues, OrcTester::reverse), filters);
+            assertRoundTrip(types, Lists.transform(readValues, OrcTester::reverse), filters, Optional.empty());
         }
 
         if (nullTestsEnabled) {
             // forward order with nulls
-            assertRoundTrip(types, insertNulls(types, readValues), filters);
+            assertRoundTrip(types, insertNulls(types, readValues), filters, optionalOrderChecker);
 
             // reverse order with nulls
             if (reverseTestsEnabled) {
-                assertRoundTrip(types, insertNulls(types, Lists.transform(readValues, OrcTester::reverse)), filters);
+                assertRoundTrip(types, insertNulls(types, Lists.transform(readValues, OrcTester::reverse)), filters, optionalOrderChecker);
             }
         }
     }
 
+    /*
+        This is a syntactic sugar for how we extended testRoundTripTypes. We added this to make the lives of the test-writer easier. (By forcing orders not to be null).
+     */
+    public void testRoundTripTypesWithOrder(List<Type> types, List<List<?>> readValues, List<Map<Integer, Map<Subfield, TupleDomainFilter>>> filters, List<Integer> order)
+            throws Exception
+    {
+        assertNotNull(order);
+        testRoundTripTypes(types, readValues, filters, Optional.of(order));
+    }
+
+    private static TupleDomainFilter addOrderTracking(TupleDomainFilter filter, OrderChecker orderChecker, int column)
+    {
+        if (filter instanceof BigintRange) {
+            OrderTrackingOrcFilter newFilter = TestBigintRange.of((BigintRange) filter, orderChecker);
+            newFilter.register(column);
+            return newFilter;
+        }
+        if (filter instanceof DoubleRange) {
+            OrderTrackingOrcFilter newFilter = TestDoubleRange.of((DoubleRange) filter, orderChecker);
+            newFilter.register(column);
+            return newFilter;
+        }
+        throw new UnsupportedOperationException("Currently, we only support checking the order of filters of type BigintRange and DoubleRange.");
```

Comment:
Include unsupported type in the message: `throw new UnsupportedOperationException("Unsupported filter type: " + filter.getClass().getSimpleName());`